app/controllers/passwords_controller.rb

```diff
@@ -26,7 +26,12 @@ def edit
     def update_from_json
       resource = resource_class.reset_password_by_token(resource_params)
       yield resource if block_given?
-      respond_to_request(resource.errors.empty?)
+      resource.valid? if resource.errors.empty?
```

Comment:
so this was confusing for me but devise `reset_password_by_token` will either return the correct instance or initiate a new empty object with a reset_token error, [see here](http://www.rubydoc.info/github/plataformatec/devise/Devise/Models/Recoverable/ClassMethods). Basically i'm using the error object as a proxy for the find by token working...